cocos/platform/CCApplicationProtocol.h

```diff
@@ -67,9 +66,6 @@ class CC_DLL ApplicationProtocol
      * @lua NA
      */
     virtual ~ApplicationProtocol(){
-#if CC_ENABLE_SCRIPT_BINDING
-        ScriptEngineManager::destroyInstance();
```

Comment:
director destructor have done it, no need to do it again. 

this is added at https://github.com/cocos2d/cocos2d-x/pull/7347 to fix memory leak